src/codegen/mod.rs

```diff
@@ -330,6 +333,27 @@ fn layout(contract_no: usize, ns: &mut Namespace) {
         }
     }
 
+    let constructors = ns.contracts[contract_no].constructors(ns);
+    if !constructors.is_empty() {
+        if let Some((_, exp)) = &ns.functions[constructors[0]].annotations.space {
+            // This code path is only reachable on Solana
+            assert_eq!(ns.target, Target::Solana);
+            if let Ok((_, value)) = eval_const_number(exp, ns, &mut Diagnostics::default()) {
+                if slot > value {
+                    ns.diagnostics.push(Diagnostic::error(
+                        exp.loc(),
+                        format!("contract requires at least {} bytes of space", slot),
+                    ));
+                } else if value > BigInt::from(10_485_760u64) {
+                    ns.diagnostics.push(Diagnostic::error(
+                        exp.loc(),
+                        "an account can be no larger than 10 MB".to_string(),
```

Comment:
this should make it clear this is restrictions of the solana runtime, not solang